File: app/src/main/java/io/github/prototypez/appjoint/app/MainActivity.java
package io.github.prototypez.appjoint.app;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;

import io.github.prototypez.appjoint.AppJoint;
import io.github.prototypez.router.FuncMonkeyRouter;
import io.github.prototypez.router.FuncMonkeyRouter2;
import io.github.prototypez.router.FuncTigerRouter;
import io.github.prototypez.router.FuncTigerRouter2;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                AppJoint.get().getRouter(FuncMonkeyRouter.class).startMonkey();
                AppJoint.get().getRouter(FuncMonkeyRouter.class).startMonkeyForResult();
                AppJoint.get().getRouter(FuncMonkeyRouter2.class).startMonkey();
                AppJoint.get().getRouter(FuncMonkeyRouter2.class).startMonkeyForResult();

                AppJoint.get().getRouter(FuncTigerRouter.class).startTiger();
                AppJoint.get().getRouter(FuncTigerRouter.class).startTigerForResult();
                AppJoint.get().getRouter(FuncTigerRouter2.class).startTiger();
                AppJoint.get().getRouter(FuncTigerRouter2.class).startTigerForResult();
            }
        });


    }

}

File: module1/src/main/java/io/github/prototypez/appjoint/module1/MainActivity.kt
package io.github.prototypez.appjoint.module1

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentPagerAdapter
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.tlHead
import kotlinx.android.synthetic.main.activity_main.vpContent

class MainActivity : AppCompatActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    vpContent.adapter = object : FragmentPagerAdapter(supportFragmentManager) {

      override fun getItem(position: Int): Fragment {
        return when (position) {
          0 -> Module1TabFragment.newInstance()
          1 -> Services.sModule2Service.module2TabFragment()
          else -> throw IllegalArgumentException("Illegal adapter index $position")
        }
      }

      override fun getCount(): Int {
        return 2
      }

      override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
          0 -> getString(R.string.module1)
          1 -> getString(R.string.module2)
          else -> throw IllegalArgumentException("Illegal adapter index $position")
        }
      }
    }

    tlHead.setupWithViewPager(vpContent)
  }
}
